### PR TITLE
[Relay][VM] Clean up the VM and VM profiler code

### PR DESCRIPTION
**Pull request by icemelon9**
- Clean up the code
- Add set_input API to VM

cc @wweic @jroesch @zhiics

### CODE REVIEW

File: include/tvm/runtime/vm.h [hunk 1]
```diff
@@ -547,6 +547,10 @@ class Executable : public ModuleNode {
    */
   runtime::Module GetLib() const { return lib; }
 
+  int GetFunctionArity(std::string func) const;
```
Comment (wweic) [inline include/tvm/runtime/vm.h]: doc string.

File: include/tvm/runtime/vm.h [hunk 2]
```diff
@@ -547,6 +547,10 @@ class Executable : public ModuleNode {
...
+
+  std::string GetFunctionParameterName(std::string func, uint32_t index) const;
```
Comment (wweic) [inline include/tvm/runtime/vm.h]: ditto.

File: tests/python/relay/test_vm.py [hunk 3]
```diff
@@ -595,7 +590,6 @@ def test_add_op_broadcast():
     test_let_scalar()
     test_compose()
     test_list_hd()
-    test_list_tl_empty_list()
```
Comment (wweic) [inline tests/python/relay/test_vm.py]: why remove the test?

Comment (icemelon9) [reply tests/python/relay/test_vm.py]: Because this one is supposed to fail.

Comment (wweic) [reply tests/python/relay/test_vm.py]: maybe it's a good idea to use `pytest.main()` in `__main__`?

**Comment by zhiics** [comment]
Thanks @icemelon9 @wweic

**Pull request merged**